.github/workflows/binaries.yml

```diff
@@ -0,0 +1,39 @@
+name: Build binaries
+on:
+  release:
+    types: [created]
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-haskell@v1
+        if: runner.os != 'Linux'
+        with:
+          ghc-version: '8.10.1'
+          cabal-version: '3.2'
+      - name: Build Linux binary
+        if: runner.os == 'Linux'
+        run: |
+          docker run \
+            -v $(pwd):/mnt -w /mnt \
+            utdemir/ghc-musl:v14-ghc8101 \
+            bash .github/workflows/build.sh
```

Comment:
This uses a different build method and in the end a different set of dependencies (Cabal-managed vs Nix-managed). Such executable is thus not tested by CI.

What I think we could do here is switching completely to GitHub actions + Cabal and pinning dependencies with Cabal freeze file. One thing that prevents this is that we now use Nix for testing extensively and a similar setup will be hard to achieve without Nix.

As an intermediate solution we could just merge this one for now.

Reply:
I don't have any experience with nix, so I can't help with that. The Windows binaries can't use Nix of course.